### PR TITLE
Branch manager panel

### PR DESCRIPTION
**Pull request by hbjORbj**
https://user-images.githubusercontent.com/32841130/234647395-89183c05-1afc-4005-9473-60f83c32cbbb.mov

- Remove Commit / Discard buttons from file/folder view
- Create a simple Branch Manager panel where we show the codemod that has just run. It has Commit / Discard buttons.

### CODE REVIEW

File: intuita-webview/src/fileExplorer/TreeItem/index.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import { ReactNode } from 'react';
+import styles from './style.module.css';
+import cn from 'classnames';
+
+type Props = {
```
Comment (grzpab) [inline intuita-webview/src/fileExplorer/TreeItem/index.tsx:5]: Please make this structure `Readonly<`

File: intuita-webview/src/fileExplorer/TreeView/index.tsx [hunk 2]
```diff
@@ -61,10 +61,10 @@ const TreeView = ({ node, fileNodes, searchQuery }: Props) => {
 	const [focusedNodeId, setFocusedNodeId] = useState('');
 
 	const handleClick = useCallback((node: TreeNode) => {
+		console.log(node, node.command);
```
Comment (grzpab) [inline intuita-webview/src/fileExplorer/TreeView/index.tsx:64]: can you remove it?

File: intuita-webview/src/campaignManager/TreeView/index.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+import { ReactComponent as WrenchIcon } from '../../assets/wrench.svg';
+import { vscode } from '../../shared/utilities/vscode';
+import styles from './style.module.css';
+import cn from 'classnames';
+
+type Props = {
+	node: TreeNode;
+};
+
+const getIcon = (iconName: string | null, open: boolean): ReactNode => {
```
Comment (grzpab) [inline intuita-webview/src/campaignManager/TreeView/index.tsx:19]: wasn't this copied copied from somewhere?

File: intuita-webview/src/campaignManager/TreeView/index.tsx [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+			break;
+	}
+	return icon;
+};
+
+const TreeView = ({ node }: Props) => {
+	const [focusedNodeId, setFocusedNodeId] = useState('');
+
+	const handleClick = useCallback((node: TreeNode) => {
+		console.log(node, node.command);
```
Comment (grzpab) [inline intuita-webview/src/campaignManager/TreeView/index.tsx:59]: can you remove this console log?

File: intuita-webview/src/fileExplorer/TreeItem/index.tsx (hunk 1 above)
Comment (hbjORbj) [reply intuita-webview/src/fileExplorer/TreeItem/index.tsx:5]: addressed

File: intuita-webview/src/fileExplorer/TreeView/index.tsx (hunk 2 above)
Comment (hbjORbj) [reply intuita-webview/src/fileExplorer/TreeView/index.tsx:64]: addressed

File: intuita-webview/src/campaignManager/TreeView/index.tsx (hunk 3 above)
Comment (hbjORbj) [reply intuita-webview/src/campaignManager/TreeView/index.tsx:19]: addressed

File: src/components/webview/CampaignManagerProvider.ts [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+			value: data,
+		});
+	}
+
+	private __postMessage(message: WebviewMessage) {
+		if (!this.__view) {
+			return;
+		}
+
+		this.__view.webview.postMessage(message);
```
Comment (grzpab) [inline src/components/webview/CampaignManagerProvider.ts:105]: `this.__view?.webview.postMessage(message)`
then the lines 101-103 are not needed

File: intuita-webview/src/campaignManager/App.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+	WebviewMessage,
+} from '../../../src/components/webview/webviewEvents';
+
+type MainViews = Extract<View, { viewId: 'campaignManagerView' }>;
+
+function App() {
+	const [view, setView] = useState<MainViews | null>(null);
+
+	useEffect(() => {
+		vscode.postMessage({ kind: 'webview.global.afterWebviewMounted' });
```
[the remark this answers is not in the document]
Comment (grzpab) [reply intuita-webview/src/campaignManager/App.tsx:17]: Put this to the end of the function block

**Review by grzpab — APPROVED** [review]

**Pull request merged**